backend/qemu.pm

```diff
@@ -418,8 +418,19 @@ sub start_qemu {
     my $qemubin = $ENV{QEMU};
 
     my $qemuimg = find_bin('/usr/bin/', qw(kvm-img qemu-img));
+
     unless ($qemubin) {
-        $qemubin = find_bin('/usr/bin/', $vars->{QEMU} ? ('qemu-system-' . $vars->{QEMU}) : qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64));
+        if ($vars->{QEMU}) {
+            $qemubin = find_bin('/usr/bin/', 'qemu-system-' . $vars->{QEMU});
+        }
+        else {
+            (my $class = $vars->{WORKER_CLASS} || '') =~ s/qemu_/qemu-system\-/g;
+            my %allowed = map { $_ => 1 } qw(kvm qemu-kvm qemu qemu-system-x86_64 qemu-system-ppc64);
+            for (split(/\s*,\s*/, $class)) {
+                $qemubin = find_bin('/usr/bin/', $_) if $allowed{$_};
+            }
+            $qemubin = find_bin('/usr/bin/', keys %allowed) unless $qemubin;
```

Comment:
this drops order - and we would run qemu-system-ppc64 before qemu by chance (perl hash randomization)

Reply:
@coolo done.